corehq/motech/generic_inbound/utils.py

```diff
@@ -14,22 +21,78 @@ def make_url_key():
     return raw_key.removesuffix("==")
 
 
-def get_context_from_request(request):
-    try:
-        body = json.loads(request.body.decode('utf-8'))
-    except (UnicodeDecodeError, json.JSONDecodeError):
-        raise GenericInboundUserError(_("Payload must be valid JSON"))
-
-    couch_user = request.couch_user
-    restore_user = couch_user.to_ota_restore_user(request.domain, request_user=couch_user)
-
-    query = dict(request.GET.lists())
-    return get_evaluation_context(
-        restore_user,
-        request.method,
-        query,
-        dict(request.headers),
-        body
+@attr.s(kw_only=True, frozen=True, auto_attribs=True)
+class ApiRequest:
+    domain: str
+    couch_user: CouchUser  # Is this correct?
```

Comment:
Is the comment still relevant?
```suggestion
    couch_user: CouchUser
```